src/internal/kopia/upload.go

```diff
@@ -531,9 +576,29 @@ func inflateCollectionTree(
 		ownerCats.ServiceCats[serviceCat] = ServiceCat{}
 		ownerCats.ResourceOwners[s.FullPath().ResourceOwner()] = struct{}{}
 
+		// Make sure we haven't moved and made a new instance of the collection at
+		// this path.
```

Comment:
This comment is kinda difficult to parse.  Does it mean that a collection in the node implies multiple collections writing data to the same path?

Reply:
if I understand your question correctly yes. Essentially this code is reached for new, not moved, and moved states. Those are the states where we have to worry about new items being added. What we want to ensure is that there's only a single collection adding items for any given path in the _new_ hierarchy

The new hierarchy is based only off the values of `FullPath()` and there are no assumptions/requirements for how collections are processed

Reply:
> ensure that there's only a single collection adding items for any given path in the new hierarchy

Aha, this ^ is what the comment in code should be.